test/units/plugins/action/test_reboot.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2022 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""Tests for the reboot action plugin."""
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+
+import pytest
+
+from ansible.playbook.task import Task
+from ansible.plugins.action.reboot import ActionModule as RebootAction
+from ansible.plugins.loader import connection_loader
+
+
+@pytest.fixture
+def task_args(request):
+    """Return playbook task args."""
+    return getattr(request, 'param', {})
+
+
+@pytest.fixture
+def module_task(mocker, task_args):
+    """Construct a task object."""
+    task = mocker.MagicMock(Task)
+    task.action = 'reboot'
+    task.args = task_args
+    task.async_val = False
+    return task
+
+
+@pytest.fixture
+def play_context(mocker):
+    """Construct a play context."""
+    ctx = mocker.MagicMock()
+    ctx.check_mode = False
+    ctx.shell = 'sh'
+    return ctx
+
+
+@pytest.fixture
+def action_plugin(play_context, module_task):
+    """Initialize an action plugin."""
+    connection = connection_loader.get('local', play_context, os.devnull)
+    loader = None
+    templar = None
+    shared_loader_obj = None
+
+    return RebootAction(
+        module_task,
+        connection,
+        play_context,
+        loader,
+        templar,
+        shared_loader_obj,
+    )
+
+
+_SENTINEL_REBOOT_COMMAND = '/reboot-command-mock --arg'
+_SENTINEL_TEST_COMMAND = 'cmd-stub'
+
+
+@pytest.mark.parametrize(
+    'task_args',
+    (
+        {
+            'reboot_timeout': 5,
+            'reboot_command': _SENTINEL_REBOOT_COMMAND,
+            'test_command': _SENTINEL_TEST_COMMAND,
+        },
+    ),
+    indirect=('task_args', ),
+)
+def test_reboot_command(action_plugin, mocker, monkeypatch):
+    """Check that the reboot command gets called and reboot verified."""
+    def _patched_low_level_execute_command(cmd, *args, **kwargs):
+        if cmd == _SENTINEL_TEST_COMMAND:
+            return {
+                'rc': 0,
+                'stderr': '<test command stub-stderr>',
+                'stdout': '<test command stub-stdout>',
+            }
+
+        if cmd == _SENTINEL_REBOOT_COMMAND:
+            return {
+                'rc': 0,
+                'stderr': '<reboot command stub-stderr>',
+                'stdout': '<reboot command stub-stdout>',
+            }
+
+        raise RuntimeError(cmd)  # This should never be hit in the test
```

Comment:
Refactoring this avoids unreachable code and gets coverage of this file to 100%:

```suggestion
        return {
            _SENTINEL_TEST_COMMAND: {
                'rc': 0,
                'stderr': '<test command stub-stderr>',
                'stdout': '<test command stub-stdout>',
            },
            _SENTINEL_REBOOT_COMMAND: {
                'rc': 0,
                'stderr': '<reboot command stub-stderr>',
                'stdout': '<reboot command stub-stdout>',
            },
        }[cmd]
```

Reply:
@mattclay I kinda wanted this to be here as a precaution for future changes. Can we do some `# pragma: no cover` here instead?

Reply:
@webknjaz What do you mean by "precaution for future changes"?  Both approaches result in an exception if `cmd` is unknown.

Use of `# pragma: nocover` should be limited to cases where there's no reasonable way to get coverage. There are a few cases where using it does make sense. A few examples (taken from our own tests):

* In `if t.TYPE_CHECKING` branches (which should also be used sparingly).
* In methods required to implement abstract types where the method raises `NotImplementedError`.

For most other uses, flow control changes (usually minor) can avoid the need for a pragma entirely.

Reply:
Agreed, you're right. With the example you posted, the exception turned from an explicit `RuntimeError` into an implicit `KeyError`, and that's what confused me initially since I'd lost some context. Honestly, I'd prefer an explicit `ThisShoudntBeHappening` but I agree that sticking pragmas and noqas everywhere is a bad idea and I tend to follow the same principle you described.

I'll adopt your suggestion.